### PR TITLE
Links and assets test

### PR DESCRIPTION
**Pull request by anibalsolon**
Following @htwangtw idea, this PR adds a test to check if all the links are working and nothing is broken.

It spawns a file server, to simulate the Github pages. Selenium spawns a headless Chrome, to navigate through all the links on the page and check that it finds every file and anchor (the "hashtag" in the URL `*dodges a rock*`).

The PR also fixes a broken image and an anchor, found by the test. Most importantly, it adds the `build` folder to `.gitignore`.

I've included some reviewers, but please feel free to review as you wish!

### CODE REVIEW

File: brainhack_book/tests/traverse_pages.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+capabilities["goog:loggingPrefs"] = {"performance": "ALL"}
+
+driver = webdriver.Chrome(options=options, desired_capabilities=capabilities)
+
+
+validate_anchors = {}
+visited_pages = set()
+to_visit_pages = [HTTP_ROOT]
+
+while len(to_visit_pages) > 0:
```
Comment (htwangtw) [inline brainhack_book/tests/traverse_pages.py:48]: I would just have:
```python
while to_visit_pages:
```
But I think how you write here is more readerable for novice python users

File: brainhack_book/tests/traverse_pages.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    page = to_visit_pages.pop(0)
+    driver.get(page)
+    visited_pages |= { page }
```
Comment (htwangtw) [inline brainhack_book/tests/traverse_pages.py:51]: Just saying this is the first time I see a project that uses sets that's not a coding challenge! 
Now I learn a practical use case of set :laughing:

**Review by htwangtw — APPROVED** [review]
I am happy with this test. Just made a few comments to show my excitement :laughing:

**Review by PeerHerholz — APPROVED** [review]
Great addition, thx a lot @anibalsolon!

**Review by SamGuay — APPROVED** [review]
I trust your skills way more than mine! :grin:

File: brainhack_book/tests/traverse_pages.py (hunk 1 above)
Comment (anibalsolon) [reply brainhack_book/tests/traverse_pages.py:48]: You are definitely correct! I am more towards the "let's make the while condition really explicit so we do not get into an infinite loop" kind of thought 🤣

File: brainhack_book/tests/traverse_pages.py (hunk 2 above)
Comment (anibalsolon) [reply brainhack_book/tests/traverse_pages.py:51]: 🥰🧑‍💻

**Comment by anibalsolon** [comment]
Thanks @htwangtw @PeerHerholz @SamGuay !!

**Pull request merged**